File: HRMS/HRMS.Web/Modules/ServerTypes/HR/DesignationRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface DesignationRow {
    DesignationId?: number;
    DesignationName?: string;
    Level?: number;
    IsActive?: boolean;
}

export abstract class DesignationRow {
    static readonly idProperty = 'DesignationId';
    static readonly nameProperty = 'DesignationName';
    static readonly localTextPrefix = 'HR.Designation';
    static readonly lookupKey = 'HR.Designation';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<DesignationRow>('HR.Designation') }
    static async getLookupAsync() { return getLookupAsync<DesignationRow>('HR.Designation') }

    static readonly deletePermission = 'HR:Designation';
    static readonly insertPermission = 'HR:Designation';
    static readonly readPermission = 'HR:Designation';
    static readonly updatePermission = 'HR:Designation';

    static readonly Fields = fieldsProxy<DesignationRow>();
}

File: HRMS/HRMS.Web/Modules/HR/Designation/DesignationRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace HRMS.HR;

[ConnectionKey("Default"), Module("HR"), TableName("Designations")]
[DisplayName("Designations"), InstanceName("Designation")]
[ReadPermission("HR:Designation")]
[ModifyPermission("HR:Designation")]
[LookupScript]
public sealed class DesignationRow : Row<DesignationRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Designation Id"), Identity, IdProperty]
    public int? DesignationId
    {
        get => fields.DesignationId[this];
        set => fields.DesignationId[this] = value;
    }

    [DisplayName("Designation Name"), Size(100), NotNull, QuickSearch, NameProperty]
    public string DesignationName
    {
        get => fields.DesignationName[this];
        set => fields.DesignationName[this] = value;
    }

    [DisplayName("Level"), NotNull]
    public int? Level
    {
        get => fields.Level[this];
        set => fields.Level[this] = value;
    }

    [DisplayName("Is Active"), NotNull]
    public bool? IsActive
    {
        get => fields.IsActive[this];
        set => fields.IsActive[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field DesignationId;
        public StringField DesignationName;
        public Int32Field Level;
        public BooleanField IsActive;
    }
}
